### PR TITLE
mark strings in javascript files for translation

### PR DESCRIPTION
**Pull request by zubair-arbi**
XCOM-343
@awais786 @aamir-khan @ahsan-ul-haq @tasawernawaz 
- Verify that all strings presented to the user via JavaScript are marked for translation.
- Include django javascript translation catalog `jsi18n` file in templates which extend from `'dashboard/layout.html'` instead of local `'edx/base.html'` (which already have this file).

### CODE REVIEW

**Comment by clintonb** [comment]
:+1:

**Comment by zubair-arbi** [comment]
@rlucioni Please review.

File: ecommerce/static/vendor-extensions/oscar/js/refund_list.js [hunk 1]
```diff
@@ -27,10 +32,15 @@ $(document).ready(function () {
             // refund, even when the refund has entered an error state during processing.
             $('tr[data-refund-id=' + refund_id + '] .refund-status').text('Error');
```
Comment (rlucioni) [inline ecommerce/static/vendor-extensions/oscar/js/refund_list.js]: This "Error" message must be marked for translation.

**Comment by zubair-arbi** [comment]
@rlucioni Please review again.

**Comment by rlucioni** [comment]
:+1: once commits are squashed and tests pass.

**Pull request merged**